### PR TITLE
[HER-15] Create user page

### PR DESCRIPTION
**Pull request by andreea-dumitrascu**
## Description
Create the user page
## Related
[HER-15](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?issueParent=10006&selectedIssue=HER-15)

### CODE REVIEW

File: src/startup/mapping.jsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { routes } from 'startup/routes'
 
 export const Mapping = () => <>
   {
-    routes.map(route => <Route exact path={route.path} key={route.path}>
+    routes.map(route => <Route path={route.path} key={route.path} exact>
```
Comment (miutamihai) [inline src/startup/mapping.jsx:8]: Was this change really necessary? :smile:

File: src/startup/routes.jsx [hunk 2]
```diff
@@ -9,4 +10,8 @@ export const routes = [{
 {
   path: '/',
   Component: HomePage,
+}, {
+  path: '/user',
```
Comment (miutamihai) [inline src/startup/routes.jsx:14]: Let's respect our convention and have the `/` route be the last in this array.

File: src/user/header/component.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+import { Grid, Paper } from '@material-ui/core'
+import PropTypes from 'prop-types'
```
Comment (miutamihai) [inline src/user/header/component.jsx:4]: Let's rearrange these imports to respect the order we've established.

File: src/user/header/index.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
```
Comment (miutamihai) [inline src/user/header/index.jsx:2]: No need for a new line here.

File: src/user/header/table-header.jsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+import { items } from 'user/header/items'
+import { Components } from 'user/header/component'
+
+export const TableHeader = () => <>
```
Comment (miutamihai) [inline src/user/header/table-header.jsx:6]: This fragment is redundant. We can remove it.

File: src/user/header/title.jsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+
+import { Grid, Paper } from '@material-ui/core'
```
Comment (miutamihai) [inline src/user/header/title.jsx:3]: Same goes here. Let's reorder our imports.

File: src/user/row/items/class.jsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+
+import { Grid, Paper } from '@material-ui/core'
```
Comment (miutamihai) [inline src/user/row/items/class.jsx:3]: Here too.

File: src/user/row/items/id.jsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+import { Grid, Paper } from '@material-ui/core'
```
Comment (miutamihai) [inline src/user/row/items/id.jsx:3]: Here too.

File: src/user/row/items/score.jsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+import { Grid, Paper } from '@material-ui/core'
```
Comment (miutamihai) [inline src/user/row/items/score.jsx:3]: Here too.

File: src/user/row/items/timestamp.jsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+import { Grid, Paper } from '@material-ui/core'
```
Comment (miutamihai) [inline src/user/row/items/timestamp.jsx:3]: I have a feeling you get what I mean by now... :smile:

File: src/user/style.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+const paper = {
+  paddingTop: '1vh',
+  paddingBottom: '1vh',
+  textAlign: 'center',
+  fontSize: '1vw',
+}
+
+const main = { maxWidth: '80%', margin: 'auto' }
```
Comment (miutamihai) [inline src/user/style.js:15]: Let's have a new line between the `main` and `useStyle` constants.

File: src/user/index.jsx [hunk 12]
```diff
@@ -0,0 +1,5 @@
+import React from 'react'
+
+import { Table } from 'user/table'
+
+export const User = () => <Table />
```
Comment (miutamihai) [inline src/user/index.jsx:5]: image
Was this `table` directory really necessary, since we only export the `Table` component?

File: src/user/table/id.jsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
+export const ids = [
```
Comment (miutamihai) [inline src/user/table/id.jsx:1]: Let's rename this file to `ids.js`

File: src/user/table/index.jsx [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+import { Header } from 'user/header'
+import { Row } from 'user/row'
+import { useStyle } from 'user/style'
+import { ids } from 'user/table/id'
+
+export const Table = () => <div className={useStyle().main}>
+  <Header />
+  {ids.map((ids, index) => <Row key={index} />)}
```
Comment (miutamihai) [inline src/user/table/index.jsx:10]: The `ids` parameter in our lambda function is not used and we use the index instead.

We can change our lambda to resemble something like this: 
```js
id => <Row key={id} />
```

Comment (miutamihai) [reply src/user/table/index.jsx:10]: Also, we can extract this map to a different component (named something like `Content`)

File: src/user/row/index.jsx [hunk 15]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+import { Grid } from '@material-ui/core'
+
+import { Id } from 'user/row/items/id'
+import { Class } from 'user/row/items/class'
+import { Score } from 'user/row/items/score'
+import { Timestamp } from 'user/row/items/timestamp'
+
+export const Row = () => <Grid container spacing={2}>
```
Comment (miutamihai) [inline src/user/row/index.jsx:10]: Should we maybe use the `id` from the array here?

**Review by miutamihai — CHANGES_REQUESTED** [review]
Only part of the review is done. Make these changes and then we'll take another look.

File: src/user/style.js (hunk 11 above)
Comment (mihairadulescu) [reply src/user/style.js:15]: and also, try to extract in separate style files for separate components.

**Review by mihairadulescu — CHANGES_REQUESTED** [review]

File: src/startup/mapping.jsx (hunk 1 above)
Comment (andreea-dumitrascu) [reply src/startup/mapping.jsx]: Done

File: src/startup/routes.jsx (hunk 2 above)
Comment (andreea-dumitrascu) [reply src/startup/routes.jsx]: Done

File: src/user/header/component.jsx (hunk 3 above)
Comment (andreea-dumitrascu) [reply src/user/header/component.jsx:3]: Done

File: src/user/header/index.jsx (hunk 4 above)
Comment (andreea-dumitrascu) [reply src/user/header/index.jsx:3]: Done

File: src/user/header/table-header.jsx (hunk 5 above)
Comment (andreea-dumitrascu) [reply src/user/header/table-header.jsx]: Done

File: src/user/header/title.jsx (hunk 6 above)
Comment (andreea-dumitrascu) [reply src/user/header/title.jsx]: Done

File: src/user/index.jsx (hunk 12 above)
Comment (andreea-dumitrascu) [reply src/user/index.jsx]: Done

File: src/user/row/index.jsx (hunk 15 above)
Comment (andreea-dumitrascu) [reply src/user/row/index.jsx]: Done

File: src/user/row/items/class.jsx (hunk 7 above)
Comment (andreea-dumitrascu) [reply src/user/row/items/class.jsx:2]: Done

File: src/user/row/items/id.jsx (hunk 8 above)
Comment (andreea-dumitrascu) [reply src/user/row/items/id.jsx:2]: Done

File: src/user/row/items/score.jsx (hunk 9 above)
Comment (andreea-dumitrascu) [reply src/user/row/items/score.jsx:2]: Done

File: src/user/row/items/timestamp.jsx (hunk 10 above)
Comment (andreea-dumitrascu) [reply src/user/row/items/timestamp.jsx:2]: Done

File: src/user/style.js (hunk 11 above)
Comment (andreea-dumitrascu) [reply src/user/style.js]: Done

File: src/user/table/id.jsx (hunk 13 above)
Comment (andreea-dumitrascu) [reply src/user/table/id.jsx]: Done

File: src/user/table/index.jsx (hunk 14 above)
Comment (andreea-dumitrascu) [reply src/user/table/index.jsx]: Done

File: src/user/header/table-header.jsx (hunk 5 above)
Comment (miutamihai) [reply src/user/header/table-header.jsx]: I did not mean to remove the component. :smile: 

I meant only the fragment (a.k.a the `<></>` part).

File: src/user/header/component.jsx [hunk 16]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { Grid, Paper } from '@material-ui/core'
+import PropTypes from 'prop-types'
+
+import { useStyle } from 'user/header/style'
+
+export const Components = ({ items }) => <Grid item xs={3}>
```
Comment (miutamihai) [inline src/user/header/component.jsx:7]: image

1. The file is named `component.jsx`, so why is the component itself named `Components`?
2. On that note, is the name `Component` really a good naming practice? I mean, aren't them all components? :smile:

File: src/user/header/index.jsx [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+import { items } from 'user/header/items'
+import { Components } from 'user/header/component'
+import { useStyle } from 'user/header/style'
+
+export const Header = () => <Grid container spacing={2}>
+  <Grid item xs={12}>
+    <Paper className={useStyle().title}>Detections</Paper>
+  </Grid>
+  {
+    items.map((items, index) => <Components items={items} key={index} />)
```
Comment (miutamihai) [inline src/user/header/index.jsx:13]: As I said [here](https://github.com/NerdsComputing/hermes.web/pull/19#discussion_r669555694), this mapping should happen in a separate component.

File: src/user/component.jsx [hunk 18]
```diff
@@ -0,0 +1,6 @@
+import React from 'react'
+
+import { ids } from 'user/id'
+import { Row } from 'user/row'
+
+export const Component = () => ids.map(ids => <Row id={ids.id} key={ids.id} />)
```
Comment (miutamihai) [inline src/user/component.jsx:6]: Same here. What does `Component` mean?

**Review by miutamihai — CHANGES_REQUESTED** [review]

File: src/user/header/table-header.jsx (hunk 5 above)
Comment (andreea-dumitrascu) [reply src/user/header/table-header.jsx]: Done

File: src/user/header/component.jsx (hunk 16 above)
Comment (andreea-dumitrascu) [reply src/user/header/component.jsx]: Done.

File: src/user/header/index.jsx (hunk 17 above)
Comment (andreea-dumitrascu) [reply src/user/header/index.jsx]: Done.

File: src/user/component.jsx (hunk 18 above)
Comment (andreea-dumitrascu) [reply src/user/component.jsx]: Done.

File: src/user/header/style.js [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+const paper = {
+  paddingTop: '1vh',
+  paddingBottom: '1vh',
+  textAlign: 'center',
+  fontSize: '1vw',
+}
+
+export const useStyle = makeStyles(() => ({ title, paper }))
```
Comment (miutamihai) [inline src/user/header/style.js:15]: Let's break up these style files as well. In that way, we can have a single inner field per style (called `root`) and thusly provide the styles to our components using the `classes` instead of the `className` prop.

File: src/user/header/header-titles.jsx [hunk 20]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { Grid, Paper } from '@material-ui/core'
+import PropTypes from 'prop-types'
+
+import { useStyle } from 'user/header/style'
+
+export const HeaderTitles = ({ items }) => <Grid item xs={3}>
```
Comment (miutamihai) [inline src/user/header/header-titles.jsx:7]: image
Since this component represents a single header, let's singularize it .

File: src/user/row/style.js [hunk 21]
```diff
@@ -0,0 +1,10 @@
+import { makeStyles } from '@material-ui/core'
+
+const paper = {
+  paddingTop: '1vh',
+  paddingBottom: '1vh',
+  textAlign: 'center',
+  fontSize: '1vw',
+}
+
+export const useStyle = makeStyles(() => ({ paper }))
```
Comment (miutamihai) [inline src/user/row/style.js:10]: Same goes here. Let's rename this field from `paper` to `root` so that we can use the `classes` prop on the components that need this style.

**Review by miutamihai — CHANGES_REQUESTED** [review]
image
Since we're probably going to add an `onClick` event to  these rows, it doesn't really make sense to have each cell split in different cards like these. Though I couldn't specifically tell you how to improve it, let's change up this design a little.

File: src/user/row/style.js (hunk 21 above)
Comment (andreea-dumitrascu) [reply src/user/row/style.js]: Done.

File: src/user/header/header-titles.jsx (hunk 20 above)
Comment (andreea-dumitrascu) [reply src/user/header/header-titles.jsx]: Done.

File: src/user/header/style.js (hunk 19 above)
Comment (andreea-dumitrascu) [reply src/user/header/style.js]: Done.

**Comment by andreea-dumitrascu** [comment]
> image
> Since we're probably going to add an `onClick` event to these rows, it doesn't really make sense to have each cell split in different cards like these. Though I couldn't specifically tell you how to improve it, let's change up this design a little.

Done.

**Review by miutamihai — CHANGES_REQUESTED** [review]
Now onto the visual stuff:
1. The `Id` column isn't really aligned with its header: 
image

2. Whilst I like the hover effect, using buttons like this makes it hard to distinguish between the different rows: 
image

Let's use cards holding the whole row, just like we discussed yesterday.

File: src/user/id.js [hunk 22]
```diff
@@ -0,0 +1,22 @@
+export const ids = [
```
[the remark this answers is not in the document]
Comment (andreea-dumitrascu) [reply src/user/id.js]: Done.

**Review by mihairadulescu — APPROVED** [review]
